Parsers/ASimDns/Parsers/vimDnsSentinelOne.yaml

```diff
@@ -0,0 +1,248 @@
+                        alertInfo_dnsResponse_s has "Unassigned",
+                        int(12),
+                        alertInfo_dnsResponse_s has "BADVERS" or alertInfo_dnsResponse_s has "Bad OPT Version",
+                        int(16),
+                        alertInfo_dnsResponse_s has "BADSIG" or alertInfo_dnsResponse_s has "TSIG Signature Failure",
+                        int(16),
+                        alertInfo_dnsResponse_s has "BADKEY" or alertInfo_dnsResponse_s has "Key not recognized",
+                        int(17),
+                        alertInfo_dnsResponse_s has "BADTIME" or alertInfo_dnsResponse_s has "Signature out of time window",
+                        int(18),
+                        alertInfo_dnsResponse_s has "BADMODE" or alertInfo_dnsResponse_s has "Bad TKEY Mode",
+                        int(19),
+                        alertInfo_dnsResponse_s has "BADNAME" or alertInfo_dnsResponse_s has "Duplicate key name",
+                        int(20),
+                        alertInfo_dnsResponse_s has "BADALG" or alertInfo_dnsResponse_s has "Algorithm not supported",
+                        int(21),
+                        alertInfo_dnsResponse_s has "BADTRUNC" or alertInfo_dnsResponse_s has "Bad Truncation",
+                        int(22),
+                        alertInfo_dnsResponse_s has "BADCOOKIE" or alertInfo_dnsResponse_s has "Bad/missing Server Cookie",
+                        int(23),
+                        int(0)
+                    ),
+          AdditionalFields = bag_pack(
+                        "MachineType",
+                        agentDetectionInfo_machineType_s,
+                        "OsRevision",
+                        agentDetectionInfo_osRevision_s
+                    )
+      | extend EventResultDetails = _ASIM_LookupDnsResponseCode(DnsResponseCode)
+      | where (responsecodename == '*' or EventResultDetails =~ responsecodename)
+      | extend 
+          DnsQueryType = iff(isempty(DnsQueryType) and DnsResponseCode == 0, int(1), DnsQueryType),
+          ThreatConfidence = coalesce(ThreatConfidence_undefined, ThreatConfidence_suspicious, ThreatConfidence_malicious)
+      | project-rename
+          EventStartTime = sourceProcessInfo_pidStarttime_t,
+          DnsQuery = alertInfo_dnsRequest_s,
+          EventUid = _ItemId,
+          DnsResponseName = alertInfo_dnsResponse_s,
+          DvcId = agentDetectionInfo_uuid_g,
+          DvcOs = agentDetectionInfo_osName_s,
+          DvcOsVersion = agentDetectionInfo_osRevision_s,
+          EventOriginalType = alertInfo_eventType_s,
+          EventOriginalSeverity = ruleInfo_severity_s,
+          EventOriginalUid = alertInfo_dvEventId_s,
+          RuleName = ruleInfo_name_s,
+          SrcProcessId = sourceProcessInfo_pid_s,
+          SrcProcessName = sourceProcessInfo_name_s,
+          SrcUsername = sourceProcessInfo_user_s,
+          ThreatOriginalConfidence = ruleInfo_treatAsThreat_s
```

Comment:
@jayeshprajapaticrest @niralishah-crest 
Hope you have validated with all available ruleInfo_treatAsThreat_s values. In the sample logs, I only had 'UNDEFINED' value

Reply:
@vakohl 
Actually we don not have the data of other type. But yes we have tested by ingesting dummy data with this values and its working as expected. Thanks